### PR TITLE
[SPARK-30896]:The behavior of JsonToStructs should not depend on SQLConf.get

### PR DESCRIPTION
**Pull request by zhulipeng**
### What changes were proposed in this pull request?
In the PR, I add the nameOfCorruptRecord parameter to the JsonToStructs expression.

### Why are the changes needed?
This allows to avoid the issue when the configuration change between different phases of planning, and this can silently break a query plan which can lead to crashes or data corruption.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By JsonExpressionsSuite.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
cc @MaxGekk

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala [hunk 1]
```diff
@@ -604,6 +608,20 @@ case class JsonToStructs(
   override def prettyName: String = "from_json"
 }
 
+object JsonToStructs{
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala]: @lipzhu Please run the linter `./dev/lint-scala` and fix the style.

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala [hunk 2]
```diff
@@ -559,7 +567,6 @@ case class JsonToStructs(
       (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getMap(0) else null
   }
 
-  val nameOfCorruptRecord = SQLConf.get.getConf(SQLConf.COLUMN_NAME_OF_CORRUPT_RECORD)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala]: `nameOfCorruptRecord` is used only in `parser` below. I would try to move it there:
```scala
@transient lazy val parser = {
  val parsedOptions = new JSONOptions(
    options,
    timeZoneId.get,
    SQLConf.get.getConf(SQLConf.COLUMN_NAME_OF_CORRUPT_RECORD))
```

**Comment by HyukjinKwon** [comment]
Closing this per the discussion in the JIRA

**Pull request closed without merging**